webapp/packages/core-executor/src/whileTask.ts

```diff
@@ -56,7 +79,7 @@ export function whileTask<T>(
         if (state) {
           resolve(value);
         } else if (!stopped) {
-          timeoutId = setTimeout(runTask, interval);
+          intervalTimeoutId = setTimeout(runTask, interval);
```

Comment:
this may lead to concurrency:
10s interval:
A -> started not ended
10s interval:
B -> started not ended

current approach is like:
10s interval:
A -> started not ended
A -> ended (after 20s for example)
10s interval:
B -> started